### ISSUE TITLE
Used variables greyed out as if unused

### ISSUE THREAD

**Issue by amks1** [microsoft/pylance-release#2406]
Issue Type: <b>Bug</b>

# Behaviour
## Expected vs. Actual

Unused variables are greyed out and used variables are white colored in my theme. However, after a while of using VSCode, all used variables get greyed out. 
![image](https://user-images.githubusercontent.com/68153785/154822324-be3986ea-8cba-4e4d-a4b3-b835db8b9855.png)

Reloading VSCode is the only way to revert back to the expected behaviour.

## Steps to reproduce:

This happens sporadically but often, and it happens suddenly after I've been coding for a while. 
I haven't found a reliable way to reproduce the error. 

# Diagnostic data

-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.10
-   Type of virtual environment used (e.g. conda, venv, virtualenv, etc.): Venv
-   Value of the `python.languageServer` setting: Default

<details>

<summary>User Settings</summary>

<p>

```

defaultLS: {"defaultLSType":"Pylance"}

downloadLanguageServer: true

envFile: "<placeholder>"

venvPath: "<placeholder>"

venvFolders: "<placeholder>"

condaPath: "<placeholder>"

pipenvPath: "<placeholder>"

poetryPath: "<placeholder>"

languageServer: "Pylance"

linting
• enabled: true
• cwd: "<placeholder>"
• Flake8Args: "<placeholder>"
• flake8Enabled: false
• flake8Path: "<placeholder>"
• lintOnSave: true
• banditArgs: "<placeholder>"
• banditEnabled: false
• banditPath: "<placeholder>"
• mypyArgs: "<placeholder>"
• mypyEnabled: false
• mypyPath: "<placeholder>"
• pycodestyleArgs: "<placeholder>"
• pycodestyleEnabled: false
• pycodestylePath: "<placeholder>"
• prospectorArgs: "<placeholder>"
• prospectorEnabled: false
• prospectorPath: "<placeholder>"
• pydocstyleArgs: "<placeholder>"
• pydocstyleEnabled: false
• pydocstylePath: "<placeholder>"
• pylamaArgs: "<placeholder>"
• pylamaEnabled: false
• pylamaPath: "<placeholder>"
• pylintArgs: "<placeholder>"
• pylintPath: "<placeholder>"

sortImports
• args: "<placeholder>"
• path: "<placeholder>"

formatting
• autopep8Args: "<placeholder>"
• autopep8Path: "<placeholder>"
• provider: "autopep8"
• blackArgs: "<placeholder>"
• blackPath: "<placeholder>"
• yapfArgs: "<placeholder>"
• yapfPath: "<placeholder>"

testing
• cwd: "<placeholder>"
• debugPort: 3000
• nosetestArgs: "<placeholder>"
• nosetestsEnabled: undefined
• nosetestPath: "<placeholder>"
• promptToConfigure: true
• pytestArgs: "<placeholder>"
• pytestEnabled: false
• pytestPath: "<placeholder>"
• unittestArgs: "<placeholder>"
• unittestEnabled: false
• autoTestDiscoverOnSaveEnabled: true

terminal
• activateEnvironment: true
• executeInFileDir: "<placeholder>"
• launchArgs: "<placeholder>"

experiments
• enabled: true
• optInto: []
• optOutFrom: []

insidersChannel: "off"

tensorBoard
• logDirectory: "<placeholder>"

```

</p>
</details>

Extension version: 2022.0.1814523869
VS Code version: Code 1.64.2 (f80445acd5a3dadef24aa209168452a3d97cc326, 2022-02-09T22:02:28.252Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz (12 x 2208)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.80GB (14.17GB free)|
|Process Argv|--crash-reporter-id a9957413-6361-4f37-8c33-4f9b5cd34d7e|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30425749
pythontb:30283811
pythonvspyt551cf:30345471
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30438340
pythondataviewer:30285071
vscod805:30301674
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30433507
vsclayoutctrc:30437038
vsrem710:30416614
dsvsc008:30440022
vsbas813:30436447
vscscmwlcmc:30438804
helix:30440343

```

</details>

**Comment by judej** [comment]
Thank for the report. Do you have any other extensions that might be interfering?

**Comment by amks1** [comment]
> Thank for the report. Do you have any other extensions that might be interfering?

I didn't face this issue before upgrading VSCode to the below version. And I haven't installed any other extensions since.
```
Version: 1.64.2 (user setup)
Commit: f80445acd5a3dadef24aa209168452a3d97cc326
Date: 2022-02-09T22:02:28.252Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19043
```

**Comment by erictraut** [comment]
I think I just ran across the same issue. By any chance, what `python.analysis.diagnosticMode` setting are you using? Are you using the default value of "openFilesOnly" or have you changed the setting to "workspace"?

**Comment by amks1** [comment]
> I think I just ran across the same issue. By any chance, what `python.analysis.diagnosticMode` setting are you using? Are you using the default value of "openFilesOnly" or have you changed the setting to "workspace"?

I'm using "openFilesOnly"

**Comment by erictraut** [comment]
I think we understand the problem and have a fix in place. If the problem you're seeing is the same one that we fixed (and I'm 90% sure it is), it will be addressed in the next release of Pylance. If the problem persists after the next release, let us know and we'll investigate further.

**Comment by amks1** [comment]
Great to know, thanks!

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.3.1, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202231-9-March-2022)
